hal/src/nRF52840/sleep_hal.cpp

```diff
@@ -133,6 +165,21 @@ static int constructBleWakeupReason(hal_wakeup_source_base_t** wakeupReason) {
     return SYSTEM_ERROR_NONE;
 }
 
+static int constructNetworkWakeupReason(hal_wakeup_source_base_t** wakeupReason, network_interface_index index) {
```

Comment:
Alternatively, we could hide this in system sleep layer and pass HAL just USART wakeupsource to make it more generalized and then on wakeup convert usart -> network.

Reply:
System sleep layer is not HAL specific. If new introduced platform didn't use USART as the interface to Cellular or Wi-Fi modem, we need to refactor it.

Reply:
Sounds reasonable.